modules/nw-configuring-qinq-sriov-proc.adoc

```diff
@@ -0,0 +1,191 @@
+----
++
+<1> This sets the S-tag VLAN tag to 171. 
+<2> Specifies the VLAN protocol to assign to the VF. Allowed values are `802.1ad` and `802.1q`. The default is `802.1q`. 
+
+.. To create the object, enter the following command:
++
+[source,terminal]
+----
+$ oc create -f nad-sriovnetwork-1ad-810.yaml
+----
+
+. Create a Network Attachment Definition with an inner VLAN. The inner VLAN is often referred to as the C-tag, or `Customer Tag`, as it belongs to the Network Function.
+
+.. Using the following YAML example, create a file named `nad-cvlan100.yaml`:
++
+[source,yaml]
+----
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: nad-cvlan100
+  namespace: default
+spec:
+  config: |
+    {
+      "cniVersion": "0.3.1",
+      "name": "vlan-100",
+      "plugins": [
+        {
+          "type": "vlan",
+          "master": "net1", <1>
+          "vlanId": 100,
+          "linkInContainer": true, 
+          "ipam": {"type": "static"}
+        }
+      ]
+    }
+----
++
+<1> Specifies the virtual function (VF) interface inside the pod. The default name is `net1` since the name is not set in the pod annotation. 
+
+.. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f nad-cvlan100.yaml
+----
+
+.Verification
+
+Verify QinQad is active on the node by following this procedure: 
+
+. Create an example test pod spec such as the following, and then save the YAML in the `test-qinq-pod.yaml` file:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+  annotations:
+    k8s.v1.cni.cncf.io/networks: sriovnetwork-1ad-810, nad-cvlan100
+spec:
+  containers:
+    - name: test-container
+      image: quay.io/openshifttest/hello-openshift:openshift
+      imagePullPolicy: Always
+      securityContext:
+        privileged: true
+----
+
+. Create the test pod by running the following command:
++
+[source,terminal]
+----
+$ oc create -f test-qinq-pod.yaml
+---- 
+
+. Obtain a list of cluster nodes: 
++
+[source,terminal]
+----
+$ oc get nodes
+----
+
+. Enter into a debug session on the target node, the node with the NIC attached. This step instantiates a debug pod called `<node_name>-debug`:
++
+[source,terminal]
+----
+$ oc debug node/my-cluster-node
+----
+
+. Set `/host` as the root directory within the debug shell. The debug pod mounts the host’s root file system in /host within the pod. By changing the root directory to /host, you can run binaries contained in the host’s executable paths:
++
+[source,terminal]
+----
+# chroot /host
+----
+
+. To display information about network interfaces, including their status and configuration, run the following command on the physical interface the virtual functions are created on:
++
+[source,terminal]
+----
+# ip -d link show ens5f0
+----
++
+.Example output
++
+[source,terminal]
+----
+7: ens5f0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
```

Comment:
this output is not correct

Reply:
ip link show ens5f0
6: ens5f0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether b4:96:91:a5:22:10 brd ff:ff:ff:ff:ff:ff
    vf 0     link/ether a2:81:ba:d0:6f:f3 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    vf 1     link/ether 8a:bb:0a:36:f2:ed brd ff:ff:ff:ff:ff:ff, vlan 171, vlan protocol 802.1ad, spoof checking on, link-state auto, trust off
    vf 2     link/ether ca:0e:e1:5b:0c:d2 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    vf 3     link/ether ee:6c:e2:f5:2c:70 brd ff:ff:ff:ff:ff:ff, vlan 171, vlan protocol 802.1ad, spoof checking on, link-state auto, trust off
    vf 4     link/ether 0a:d6:b7:66:5e:e8 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    vf 5     link/ether da:d5:e7:14:4f:aa brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    vf 6     link/ether d6:8e:85:75:12:5c brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    vf 7     link/ether d6:eb:ce:9c:ea:78 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off
    vf 8     link/ether 5e:c5:cc:05:93:3c brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust on
    vf 9     link/ether a6:5a:7c:1c:2a:16 brd ff:ff:ff:ff:ff:ff, spoof checking on, link-state auto, trust off